docs/configuration_reference.rst
Configuration reference
=======================

All available configuration options are listed below with their default values.

.. code-block:: yaml

    nucleos_user:
        firewall_name:          ~ # Required
        user_class:             ~ # Required
        use_flash_notifications:    true
        model_manager_name:         null  # change it to the name of your entity/document manager if you don't want to use the default one.
        use_authentication_listener: true
        from_email:             ~ # Required
        resetting:
            retry_ttl: 7200 # Value in seconds, logic will use as hours
            token_ttl: 86400
            from_email: # Use this node only if you don't want the global email address for the resetting email
        service:
            mailer:                 nucleos_user.mailer.simple
            token_generator:        nucleos_user.util.token_generator.simple
            user_manager:           nucleos_user.user_manager.default
        group:
            group_class:    ~ # Required when using groups
            group_manager:  nucleos_user.group_manager.default
        loggedin:
            route: ~ # Required


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file, in reverse chronological order by release.

## 3.4.4 - TBD

### Added

- Nothing.

### Changed

- Nothing.

### Deprecated

- Nothing.

### Removed

- Nothing.

### Fixed

- Nothing.

## 3.4.3 - 2024-07-07


-----

### Release Notes for [3.4.3](https://github.com/nucleos/NucleosUserBundle/milestone/44)

3.4.x bugfix release (patch)

### 3.4.3

- Total issues resolved: **0**
- Total pull requests resolved: **1**
- Total contributors: **1**

#### Bug

 - [759: Fix setting locale](https://github.com/nucleos/NucleosUserBundle/pull/759) thanks to @core23

## 3.4.2 - 2024-07-07


-----

### Release Notes for [3.4.2](https://github.com/nucleos/NucleosUserBundle/milestone/43)

3.4.x bugfix release (patch)

### 3.4.2

- Total issues resolved: **0**
- Total pull requests resolved: **1**
- Total contributors: **1**

#### Bug

 - [757: Fix unsetting locale](https://github.com/nucleos/NucleosUserBundle/pull/757) thanks to @core23

## 3.4.1 - 2024-05-20


-----

### Release Notes for [3.4.1](https://github.com/nucleos/NucleosUserBundle/milestone/41)

3.4.x bugfix release (patch)

### 3.4.1

- Total issues resolved: **0**
- Total pull requests resolved: **2**
- Total contributors: **1**

#### Bug

 - [754: Fix translation of auth provider messages](https://github.com/nucleos/NucleosUserBundle/pull/754) thanks to @core23

#### Enhancement

 - [751: Add functional tests for all actions](https://github.com/nucleos/NucleosUserBundle/pull/751) thanks to @core23

## 3.4.0 - 2024-03-06


-----

### Release Notes for [3.4.0](https://github.com/nucleos/NucleosUserBundle/milestone/38)

Feature release (minor)

### 3.4.0

- Total issues resolved: **0**
- Total pull requests resolved: **1**
- Total contributors: **1**

#### Enhancement

 - [750: Deprecate user plainPassword in favor of using UserManipulator](https://github.com/nucleos/NucleosUserBundle/pull/750) thanks to @core23

## 3.3.2 - 2024-03-06


-----

### Release Notes for [3.3.2](https://github.com/nucleos/NucleosUserBundle/milestone/39)

3.3.x bugfix release (patch)

### 3.3.2

- Total issues resolved: **0**
- Total pull requests resolved: **1**
- Total contributors: **1**

#### Bug

 - [748: Reset passwordRequestedAt on plain password change to trigger doctrine listener](https://github.com/nucleos/NucleosUserBundle/pull/748) thanks to @core23

## 3.3.1 - 2024-03-04


-----

### Release Notes for [3.3.1](https://github.com/nucleos/NucleosUserBundle/milestone/37)

3.3.x bugfix release (patch)

### 3.3.1

- Total issues resolved: **0**
- Total pull requests resolved: **1**
- Total contributors: **1**

#### Bug

 - [746: Revert: Fix roles doctrine type mapping](https://github.com/nucleos/NucleosUserBundle/pull/746) thanks to @core23

## 3.3.0 - 2024-03-04


-----

### Release Notes for [3.3.0](https://github.com/nucleos/NucleosUserBundle/milestone/35)

Feature release (minor)

### 3.3.0

- Total issues resolved: **0**
- Total pull requests resolved: **1**
- Total contributors: **1**

#### Bug

 - [745: Fix roles doctrine type mapping](https://github.com/nucleos/NucleosUserBundle/pull/745) thanks to @core23

## 3.2.1 - 2024-02-07


-----

### Release Notes for [3.2.1](https://github.com/nucleos/NucleosUserBundle/milestone/34)

3.2.x bugfix release (patch)

### 3.2.1

- Total issues resolved: **0**
- Total pull requests resolved: **1**
- Total contributors: **1**

#### Bug

- [742: Fix doctrine listener registration](https://github.com/nucleos/NucleosUserBundle/pull/742) thanks to @core23

## 3.2.0 - 2024-02-04


-----

### Release Notes for [3.2.0](https://github.com/nucleos/NucleosUserBundle/milestone/33)

Feature release (minor)

### 3.2.0

- Total issues resolved: **0**
- Total pull requests resolved: **5**
- Total contributors: **3**

#### dependency

 - [741: Update dependency doctrine/orm to ^2.18 || ^3.0](https://github.com/nucleos/NucleosUserBundle/pull/741) thanks to @core23
 - [733: Bump to symfony ^6.4 || ^7.0](https://github.com/nucleos/NucleosUserBundle/pull/733) thanks to @core23
 - [729: Update dependency matthiasnoback/symfony-dependency-injection-test to v5](https://github.com/nucleos/NucleosUserBundle/pull/729) thanks to @renovate[bot]

#### Enhancement

 - [732: Update tools ](https://github.com/nucleos/NucleosUserBundle/pull/732) thanks to @core23

#### Bug

 - [728: Fix login form uses getMessage() instead of getMessageKey()](https://github.com/nucleos/NucleosUserBundle/pull/728) thanks to @pohlm01

## 3.1.0 - 2023-10-01


-----

### Release Notes for [3.1.0](https://github.com/nucleos/NucleosUserBundle/milestone/27)

Feature release (minor)

### 3.1.0

- Total issues resolved: **0**
- Total pull requests resolved: **2**
- Total contributors: **1**

#### Enhancement

 - [723: Deprecate CheckEmailAction in favor of flash messages](https://github.com/nucleos/NucleosUserBundle/pull/723) thanks to @core23

#### Bug

 - [716: Remove password constraint from &quot;User&quot; group](https://github.com/nucleos/NucleosUserBundle/pull/716) thanks to @core23

## 3.0.3 - 2023-09-09


-----

### Release Notes for [3.0.3](https://github.com/nucleos/NucleosUserBundle/milestone/30)

3.0.x bugfix release (patch)

### 3.0.3

- Total issues resolved: **0**
- Total pull requests resolved: **1**
- Total contributors: **1**

#### Bug

 - [721: Prevent account detection](https://github.com/nucleos/NucleosUserBundle/pull/721) thanks to @core23

## 3.0.2 - 2023-08-23


-----

### Release Notes for [3.0.2](https://github.com/nucleos/NucleosUserBundle/milestone/29)

3.0.x bugfix release (patch)

### 3.0.2

- Total issues resolved: **0**
- Total pull requests resolved: **1**
- Total contributors: **1**

#### Bug

 - [719: Fix routing](https://github.com/nucleos/NucleosUserBundle/pull/719) thanks to @core23

## 3.0.1 - 2023-08-20


-----

### Release Notes for [3.0.1](https://github.com/nucleos/NucleosUserBundle/milestone/26)

3.0.x bugfix release (patch)

### 3.0.1


src/DependencyInjection/Configuration.php
<?php

declare(strict_types=1);

/*
 * This file is part of the NucleosUserBundle package.
 *
 * (c) Christian Gripp <mail@core23.de>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

namespace Nucleos\UserBundle\DependencyInjection;

use Symfony\Component\Config\Definition\Builder\NodeDefinition;
use Symfony\Component\Config\Definition\Builder\TreeBuilder;
use Symfony\Component\Config\Definition\ConfigurationInterface;

final class Configuration implements ConfigurationInterface
{
    public function getConfigTreeBuilder(): TreeBuilder
    {
        $treeBuilder = new TreeBuilder('nucleos_user');

        $rootNode = $treeBuilder->getRootNode();

        $this->addMainSection($rootNode);
        $this->addResettingSection($rootNode);
        $this->addDeletionSection($rootNode);
        $this->addGroupSection($rootNode);
        $this->addServiceSection($rootNode);
        $this->addLoggedinSection($rootNode);

        return $treeBuilder;
    }

    private function addMainSection(NodeDefinition $node): void
    {
        $node
            ->children()
                ->scalarNode('user_class')->isRequired()->cannotBeEmpty()->end()
                ->scalarNode('firewall_name')->isRequired()->cannotBeEmpty()->end()
                ->scalarNode('model_manager_name')->defaultNull()->end()
                ->booleanNode('use_authentication_listener')->defaultTrue()->end()
                ->booleanNode('use_flash_notifications')->defaultTrue()->end()
                ->scalarNode('from_email')->isRequired()->cannotBeEmpty()->end()
            ->end()
        ;
    }

    private function addResettingSection(NodeDefinition $node): void
    {
        $node
            ->children()
                ->arrayNode('resetting')
                    ->addDefaultsIfNotSet()
                    ->canBeUnset()
                    ->children()
                        ->scalarNode('retry_ttl')->defaultValue(7200)->end()
                        ->scalarNode('token_ttl')->defaultValue(86400)->end()
                        ->scalarNode('from_email')->end()
                    ->end()
                ->end()
            ->end()
        ;
    }

    private function addDeletionSection(NodeDefinition $node): void
    {
        $node
            ->addDefaultsIfNotSet()
            ->children()
                ->arrayNode('deletion')
                    ->canBeEnabled()
                ->end()
            ->end()
        ;
    }

    private function addServiceSection(NodeDefinition $node): void
    {
        $node
            ->addDefaultsIfNotSet()
            ->children()
                ->arrayNode('service')
                    ->addDefaultsIfNotSet()
                        ->children()
                            ->scalarNode('mailer')->defaultValue('nucleos_user.mailer.simple')->end()
                            ->scalarNode('token_generator')->defaultValue('nucleos_user.util.token_generator.simple')->end()
                            ->scalarNode('user_manager')->defaultValue('nucleos_user.user_manager.default')->end()
                        ->end()
                    ->end()
                ->end()
            ->end()
        ;
    }

    private function addGroupSection(NodeDefinition $node): void
    {
        $node
            ->children()
                ->arrayNode('group')
                    ->canBeUnset()
                    ->children()
                        ->scalarNode('group_class')->isRequired()->cannotBeEmpty()->end()
                        ->scalarNode('group_manager')->defaultValue('nucleos_user.group_manager.default')->end()
                    ->end()
                ->end()
            ->end()
        ;
    }

    private function addLoggedinSection(NodeDefinition $node): void
    {
        $node
            ->addDefaultsIfNotSet()
            ->children()
                ->arrayNode('loggedin')
                    ->addDefaultsIfNotSet()
                    ->children()
                        ->scalarNode('route')->isRequired()->cannotBeEmpty()->end()
                    ->end()
                ->end()
            ->end()
        ;
    }
}
